packages/indexer/src/vcr/client.ts

```diff
@@ -0,0 +1,73 @@
+import type {
+  Client,
+  ClientCallOptions,
+  StatusRequest,
+  StreamConfig,
+  StreamDataOptions,
+  StreamDataRequest,
+  StreamDataResponse,
+} from "@apibara/protocol";
+import type { VcrConfig } from "./config";
+import path from "node:path";
+import fs from "node:fs";
+import type { CassetteDataType } from "./record";
+import assert from "node:assert";
+
+export class VcrClient<TFilter, TBlock> implements Client<TFilter, TBlock> {
+  constructor(
+    private vcrConfig: VcrConfig,
+    private cassetteName: string,
+    private streamConfig: StreamConfig<TFilter, TBlock>,
+  ) {}
+
+  async status(request?: StatusRequest, options?: ClientCallOptions) {
```

Comment:
I don't think status makes any sense for the vcr. Throw an error like `"Client.status is not implemented for VcrClient"`.